File: MindSync-VR/app/src/main/java/com/mindsyncvr/core/data/MockData.kt
package com.mindsyncvr.core.data

import com.mindsyncvr.core.model.*

object MockData {
    val demoUser = UserProfile(
        id = "user_demo_001",
        email = "participant@mindsync.local",
        name = "Ari"
    )

    val devices = listOf(
        WearableDevice("ble-aurora-01", "MindSync Band A1", -48, 84, "1.8.2", "Today", 91, 88),
        WearableDevice("ble-calm-02", "BioSense Calm Patch", -66, 62, "0.9.7", "Today", 76, 79),
        WearableDevice("ble-lab-03", "Lab HRV Sensor", -72, 51, "2.1.0", "Yesterday", 69, 72)
    )

    val researchState = ResearchComponentState(
        signalConfidence = 88,
        sensorQuality = "Stable PPG + HRV window",
        stressLevel = 34,
        stressBand = StressBand.Balanced,
        stressSummary = "Balanced with mild activation",
        vrAdaptationState = "Softening visual density",
        environmentProfile = "Ocean dusk",
        personalizationStatus = "Matched to low-motion preference",
        validationPending = true,
        validationCompletion = "Not started",
        audioPersonalizationActive = true,
        soundAdaptationLevel = 42,
        ambientBlendingState = "Warm pads with low nature layer",
        therapeuticAudioMode = "Guided breath pacing"
    )

    val sessions = listOf(
        MeditationSession("s-104", "Ocean Dusk Reset", 18, "Ocean", "Warm pads", 100, 4, 8, true, "Good relaxation response, no discomfort reported."),
        MeditationSession("s-103", "Forest Grounding", 12, "Forest", "Nature heavy", 96, 5, 7, true),
        MeditationSession("s-102", "Abstract Calm Focus", 10, "Abstract calm", "Soft drone", 93, 6, 7, false)
    )

    val questionnaires = listOf(
        QuestionnaireTemplate(
            id = "pre-stress-v1",
            title = "Pre-session check-in",
            description = "A short baseline before your meditation begins.",
            component = "pre_session",
            version = "1.0",
            questions = listOf(
                QuestionnaireQuestion("stress_now", "How much stress do you notice right now?", QuestionType.Slider, required = true, min = 0, max = 10),
                QuestionnaireQuestion("mood_now", "How would you rate your current mood?", QuestionType.Likert, required = true, min = 1, max = 7),
                QuestionnaireQuestion("body_state", "Which body state feels most present?", QuestionType.SingleChoice, required = true, options = listOf("Settled", "Tense", "Tired", "Restless", "Neutral"))
            )
        ),
        QuestionnaireTemplate(
            id = "component-d-post-v1",
            title = "Post-session validation",
            description = "Component D validation for relaxation, immersion, safety, and perceived benefit.",
            component = "component_d",
            version = "1.0",
            questions = listOf(
                QuestionnaireQuestion("relaxation_after", "How relaxed do you feel now?", QuestionType.Slider, required = true, min = 0, max = 10),
                QuestionnaireQuestion("immersion", "How immersive did the VR environment feel?", QuestionType.Likert, required = true, min = 1, max = 7),
                QuestionnaireQuestion("audio_fit", "The audio felt supportive for my current state.", QuestionType.Likert, required = true, min = 1, max = 7),
                QuestionnaireQuestion("discomfort", "Did you experience discomfort?", QuestionType.SingleChoice, required = true, options = listOf("No", "Mild", "Moderate", "Strong")),
                QuestionnaireQuestion("discomfort_detail", "What discomfort did you notice?", QuestionType.MultipleChoice, options = listOf("Dizziness", "Eye strain", "Audio sensitivity", "Emotional discomfort", "Motion sensitivity"), branch = BranchRule("discomfort", equals = "Moderate")),
                QuestionnaireQuestion("notes", "Anything you would like the research team to understand?", QuestionType.Text)
            )
        ),
        QuestionnaireTemplate(
            id = "adverse-report-v1",
            title = "Discomfort report",
            description = "Use this if anything felt unsafe, too intense, or uncomfortable.",
            component = "adverse",
            version = "1.0",
            questions = listOf(
                QuestionnaireQuestion("severity", "How intense was the experience?", QuestionType.SingleChoice, required = true, options = listOf("Mild", "Moderate", "Strong")),
                QuestionnaireQuestion("support_needed", "Would you like a follow-up from the study team?", QuestionType.SingleChoice, required = true, options = listOf("Yes", "No")),
                QuestionnaireQuestion("description", "Describe what happened in your own words.", QuestionType.Text, required = true)
            )
        )
    )
}

File: MindSync-VR/react-native-app/src/constants/mockData.ts
import {MeditationSession, QuestionnaireTemplate, UserProfile} from '../types/domain';

export const demoUser: UserProfile = {
  id: 'participant-demo', email: 'ari@mindsync.study', name: 'Ari', role: 'participant',
  onboardingComplete: true, preferredLanguage: 'English',
};

export const demoSessions: MeditationSession[] = [
  {id: 's-104', title: 'Ocean Dusk Reset', date: '2026-08-28', durationMinutes: 15, environment: 'Ocean', audioProfile: 'Warm pads', completionRate: 100, moodBefore: 5, moodAfter: 8, validationComplete: false},
  {id: 's-103', title: 'Forest Focus', date: '2026-08-25', durationMinutes: 12, environment: 'Forest', audioProfile: 'Nature heavy', completionRate: 100, moodBefore: 4, moodAfter: 7, validationComplete: true},
  {id: 's-102', title: 'Mountain Stillness', date: '2026-08-22', durationMinutes: 10, environment: 'Mountain', audioProfile: 'Soft drone', completionRate: 92, moodBefore: 6, moodAfter: 8, validationComplete: true},
];

export const questionnaireTemplates: QuestionnaireTemplate[] = [
  {
    id: 'component-d-post-v1', title: 'Post-session validation',
    description: 'A brief reflection linked to your latest meditation session.', component: 'D', version: '1.0.0',
    questions: [
      {id: 'relaxation', prompt: 'How relaxed do you feel now?', type: 'likert', required: true, min: 1, max: 7},
      {id: 'immersion', prompt: 'How immersed did you feel in the VR environment?', type: 'slider', required: true, min: 0, max: 10},
      {id: 'discomfort', prompt: 'Did you experience any discomfort?', type: 'single', required: true, options: ['No', 'Mild', 'Moderate', 'Severe']},
      {id: 'discomfort-note', prompt: 'Please describe what felt uncomfortable.', type: 'text', branch: {whenQuestionId: 'discomfort', includes: 'Moderate'}},
    ],
  },
  {
    id: 'pre-session-v1', title: 'Pre-session check-in', description: 'A private baseline for this session.', component: 'B', version: '1.0.0',
    questions: [
      {id: 'stress', prompt: 'How activated or stressed do you feel?', type: 'slider', required: true, min: 0, max: 10},
      {id: 'mood', prompt: 'How would you describe your mood?', type: 'single', required: true, options: ['Low', 'Uneasy', 'Neutral', 'Steady', 'Positive']},
    ],
  },
];
